llmfoundry/callbacks/hf_checkpointer.py

```diff
@@ -107,32 +112,82 @@ def _maybe_get_license_filename(
         return None
 
 
-def _register_model_with_run_id_multiprocess(
+def _log_model_multiprocess(
     mlflow_logger: MLFlowLogger,
-    composer_logging_level: int,
-    model_uri: str,
-    name: str,
+    python_logging_level: int,
+    transformers_model_path: str,
+    flavor: str,
+    input_example: dict[str, Any],
+    log_model_metadata: dict[str, str],
+    task: str,
     await_creation_for: int,
+    registered_model_name: Optional[str] = None,
 ):
-    """Call MLFlowLogger.register_model_with_run_id.
-
-    Used mainly to register from a child process.
+    """
+    Call MLFlowLogger.log_model.
+    
+    Used mainly to log from a child process.
+
+    Inputs:
+    - mlflow_logger: MLFlowLogger: MLflow logger object
+    - python_logging_level: int: logging level
+    - flavor: str: transformers or peft
+    - input_example: dict[str, Any]: model serving input example for model
+    - log_model_metadata: dict[str, str]: This metadata is currently needed for optimized serving
+    - task: str: LLM task for model deployment (i.e. chat or completions)
+    - await_creation_for: int: time to wait for model creation
+    - registered_model_name: Optional
     """
     # Setup logging for child process. This ensures that any logs from composer are surfaced.
-    if composer_logging_level > 0:
+    if python_logging_level > 0:
         # If logging_level is 0, then the composer logger was unset.
         logging.basicConfig(
             format=
             f'%(asctime)s: rank{dist.get_global_rank()}[%(process)d][%(threadName)s]: %(levelname)s: %(name)s: %(message)s',
         )
-        logging.getLogger('composer').setLevel(composer_logging_level)
-
-    # Register model.
-    mlflow_logger.register_model_with_run_id(
-        model_uri=model_uri,
-        name=name,
-        await_creation_for=await_creation_for,
+        logging.getLogger('llmfoundry').setLevel(python_logging_level)
+    
+    # monkey patch to prevent duplicate tokenizer upload
+    import mlflow
+    mlflow.start_run(
+        run_id=mlflow_logger._run_id,
     )
+    original_save_model = mlflow.transformers.save_model
+    def save_model_patch(*args: Any, **kwargs: Any):
+        original_save_model(*args, **kwargs)
+        tokenizer_files = []
+        # Check if there are duplicate tokenizer files in the model directory and remove them.
+        try:
+            for tokenizer_file_name in tokenizer_files:
+                dupe_file = os.path.isfile(os.path.join(kwargs['path'], 'model', tokenizer_file_name))
+                if dupe_file:
+                    log.debug(f"Removing duplicate tokenizer file: {tokenizer_file_name}")
+                    os.remove(os.path.join(kwargs['path'], 'model', tokenizer_file_name))
+        except Exception as e:
+            log.error(f"Exception when removing duplicate tokenizer files in the model directory", e)
+    mlflow.transformers.save_model = save_model_patch
+
+    if registered_model_name is not None:
```

Comment:
you don't need a big if/else here. presumably log_model is fine with you passing `registered_model_name=None`